src/com/amazon/ion/impl/lite/IonValueLite.java

```diff
@@ -651,7 +651,9 @@ private boolean clearSymbolIDsIterative(boolean readOnlyMode) {
                 if (++clearSymbolIDsHolderStackIndex >= clearSymbolIDsHolderStack.length) {
                     clearSymbolIDsHolderStack = growClearSymbolIDsHolderStack(clearSymbolIDsHolderStack);
                 }
-                currentClearSymbolIDsHolder = new ClearSymbolIDsHolder();
+                if (clearSymbolIDsHolderStack[clearSymbolIDsHolderStackIndex] == null) {
+                    currentClearSymbolIDsHolder = new ClearSymbolIDsHolder();
+                }
```

Comment:
Should this be:

```suggestion
                currentClearSymbolIDsHolder = clearSymbolIDsHolderStack[clearSymbolIDsHolderStackIndex];
                if (currentClearSymbolIDsHolder == null) {
                    currentClearSymbolIDsHolder = new ClearSymbolIDsHolder();
                }
```

Did the tests pass without this? It looks like you'd end up using the ClearSymbolIDsHolder from the previous stack index, but maybe I'm missing something.

Reply:
Yes, it did. Maybe because irrespective of if we add your suggested line 654, we will still set all the fields of `currentClearSymbolIdsHolder` to current value after the if statements(Line 657-662). So it will always be changed to current value. But that being said my solution doesn't garbage collect the iterator properly it might make sense to do it this way.